src/object/cli_ec.c

```diff
@@ -259,30 +337,29 @@ ec_update_params(struct ec_params *params, daos_iod_t *iod, d_sg_list_t *sgl,
 		if (stripe_cnt > 1) {
 			D_REALLOC_ARRAY(nrecx,
 					(niod->iod_recxs),
-					(niod->iod_nr +
-					 iod->iod_nr + stripe_cnt - 1));
+					(niod->iod_nr + iod->iod_nr +
+					 partial_cnt + stripe_cnt - 1));
 			if (nrecx == NULL) {
 				D_FREE(niod->iod_recxs);
 				return -DER_NOMEM;
 			}
 			niod->iod_recxs = nrecx;
 		}
+		D_ASSERT(rem > 0);
 		while (rem) {
-			uint32_t stripe_len = (len * k) / iod->iod_size;
-
-			if (rem <= len * k) {
-				niod->iod_recxs[params->niod.iod_nr].
-				rx_nr = rem/iod->iod_size;
-				niod->iod_recxs[params->niod.iod_nr++].
-				rx_idx = start;
+			if (rem <= ss) {
+				niod->iod_recxs[params->niod.iod_nr].rx_nr =
+					rem/iod->iod_size;
+				niod->iod_recxs[params->niod.iod_nr++].rx_idx =
+					start/iod->iod_size;
```

Comment:
is iod_size guaranteed to be known at this point?

Reply:
I think our assumption here is that EC requires array value to pass record size?

Reply:
This version of the code does everything in bytes, but the IOD expects records.  Here, we're converting back to records.  The iod_size is guaranteed to be known.